### ISSUE TITLE
[Bug] partition_by config parameter doesn't work with Snowflake

### ISSUE THREAD

**Issue by jasongroob** [fivetran/dbt_mixpanel#37]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

In the `mixpanel__<table>` data models, the config block is set-up to use the `partition_by` which is not a recognized parameter for snowflake.  For clustering to work in Snowflake, you need to use the `cluster_by` parameter.

The current value of `partition_by={'field': 'date_day', 'data_type': 'date'} if target.type not in ('spark','databricks') else ['date_day'],` has some logic that does use different values for databricks vs all other platforms.  The issue is that Snowflake just ignores this parameter completely.

To fix this issue, I followed the [instruction](https://community.fivetran.com/t5/user-group-for-dbt/overriding-fivetran-dbt-package-default-models/td-p/517) of disabling the base `dbt_mixpanel` version of the model and writing a copy of the model with the correct config block.

Here's the config block that ended up working for me.  Without having other platforms to test, I do not know if this would work.  I assume Databricks needs partition_by while Snowflake needs cluster_by.
```
{{
    config(
        materialized='incremental',
        unique_key='unique_event_id',
        cluster_by="date_day",
        incremental_strategy = 'delete+insert',
    )
}}
```

### Relevant error log or model output

```shell
Using the default parameters on Snowflake, this the last few lines of the of the compiled SQL.



    from dedupe

)

select * from pivot_properties

```
```


### Expected behavior

Using the correct Snowflake parameters, this is the last few lines of the compile SQL.
```
        

    from dedupe

)

select * from pivot_properties
              ) order by (date_day)
        );
      alter table dbt.int_mixpanel__event cluster by (date_day);
  
```

### dbt Project configurations

config-version: 2
name: 'mixpanel'
version: '0.8.0'
require-dbt-version: [">=1.3.0", "<2.0.0"]

### Package versions

  - package: fivetran/mixpanel
    version: 0.8.0

### What database are you using dbt with?

snowflake

### dbt Version

- installed: 1.6.2

### Additional Context

_No response_

### Are you willing to open a PR to help address this issue?

- [ ] Yes.
- [ ] Yes, but I will need assistance and will schedule time during our [office hours](https://calendly.com/fivetran-solutions-team/fivetran-solutions-team-office-hours) for guidance
- [X] No.

**Comment by fivetran-catfritz** [comment]
Hi @jasongroob thank you for opening this issue. I am looking into if there is a way to use partitioning vs clustering depending on the target warehouse. Out of curiosity, what would you say the performance gain is when specifying a c`luster by` field in Snowflake? Reading [this doc](https://docs.getdbt.com/reference/resource-configs/snowflake-configs), it mentions that snowflake has automatic clustering and that specifying a `cluster by` fields reduces the workload, so wondering what kind of benefit this is for our Snowflake users? Thanks!

**Comment by jasongroob** [comment]
Hi @fivetran-catfritz, thanks for looking into this!

It looks like automatic_clustering is disabled on our snowflake instance.  Not sure how or why that happened.  I'll ask around internally.

Regarding performance, we were storing a few years of mixpanel data and every query was doing a full table scan.  Since enabling `cluster_by`, the behavior is as expected with the table actually being clustered.  I don't have any hard stats around the improvement, but simple queries went from minutes to seconds.

**Comment by jasongroob** [comment]
I did some more digging.  Snowflake automatic clustering is a feature that works with already clustered tables.  It doesn't automatically cluster data based on what it thinks will be most optimal.

The first line of the [Snowflake documentation](https://docs.snowflake.com/en/user-guide/tables-auto-reclustering) says: 

> Automatic Clustering is the Snowflake service that seamlessly and continually manages all reclustering, as needed, of clustered tables.

As it relates to this mixpanel package, the data is unclustered by default.  It's only enabled when the `cluster_by` parameter is passed in.

So using the partition_by parameter results in an unclustered table that performs full table scans.  Adding cluster_by allows more efficient querying when filtering by date.

**Comment by fivetran-catfritz** [comment]
Hi @jasongroob thank you much for the extra info! It definitely seems like a very useful thing to add. Also good news it seems like all we have to do is add an additional line for the `cluster_by` setting to our configs. Though it would help if you could confirm that having both a `partion_by` and a `cluster_by` setting doesn't cause any issues for you. I went ahead and added this config line to the models in a test branch.

You can install it using the below snippet in your `packages.yml` in place of the current mixpanel package. 
```yml
- git: https://github.com/fivetran/dbt_mixpanel.git
  revision: test/snowflake-cluster-by
  warn-unpinned: false
```

If you are able to try it out, does that work as expected?

**Comment by jasongroob** [comment]
hi @fivetran-catfritz, i pulled down the dev version of dbt_mixpanel and it is working great.  It ran fine on Snowflake and created a clustered table.

I verified this by looking at the dbt compiled code (i.e. it included `cluster by (date_day)`) and by checking the snowflake metadata on the built code.

thanks for making these updates!

**Comment by fivetran-catfritz** [comment]
@jasongroob Thanks for trying it out, and glad to hear that worked. We do plan to include this update in a future sprint, though we don't have a definite timeline yet. In the meantime I won't delete the test branch, so you can continue to use that if you wish!

**Comment by jasongroob** [comment]
@fivetran-catfritz no rush on our end!  I've turned off the default `mixpanel__events` model and rebuilt it in our project with the appropriate cluster_by flag.

**Comment by fivetran-catfritz** [comment]
Hello @jasongroob, I am working on enhancing the overall performance of this Mixpanel package and have added additional updates on a test branch. I've made adjustments to some of the clustering columns and updated the incremental strategies for all the models. In addition, I've eliminated the `stg_mixpanel__event_tmp` model in favor of transforming `stg_mixpanel__event` into an incremental model. 

If you would be able to try out the test branch and let me know your thoughts and feedback, it would be appreciated! I am curious to know how much of an impact these updates might make for you. Thank you!

To install the test branch, you can use the following in your packages.yml:
```yml
packages:
  - git: https://github.com/fivetran/dbt_mixpanel.git
    revision: feature/performance-enhancement
    warn-unpinned: false
```

**Comment by jasongroob** [comment]
@fivetran-catfritz thanks for making these updates! I'll take a look soon and let you know if I have any comments

**Comment by fivetran-catfritz** [comment]
Hi @jasongroob thanks again for your insightful comments on my PR. They definitely helped me think through my updates. I have gone ahead and released [v0.9.0](https://github.com/fivetran/dbt_mixpanel/releases/tag/v0.9.0) and will mark this issue as closed, however if you have any additional feedback, please feel free to tag us in this thread or open a new issue!
